module/zfs/zio.c

```diff
@@ -526,17 +524,18 @@ zio_decrypt(zio_t *zio, abd_t *data, uint64_t size)
 			 * from the indirect block. We decompress it now and
 			 * throw away the result after we are finished.
 			 */
-			tmp = zio_buf_alloc(lsize);
+			abd_t *abd = abd_alloc(lsize, B_TRUE);
```

Comment:
I think `abd_alloc_linear()` could be better here to make sure we won't get scatter ABD and do unnecessary copying.

Reply:
Oh yeah, that's an actual behaviour change hey. Good grab, fixed.